### PR TITLE
sql: propagate bytea_output to distsql processors

### LINKED ISSUE #27950: distsql: sessiondata fields are not properly initialized for processors
The BytesEncodeFormat field is not properly propagated.

looking into it

cc @jordanlewis

### PR DESCRIPTION
**Pull request by knz**
Fixes  #27950.

It was not propagated previously, causing distributed queries to
improperly convert byte arrays to strings.

Release note (bug fix): the parameter `bytea_output` is now properly
effective for distributed queries.

### CODE REVIEW

**Review by BramGruneir — APPROVED** [review]
:lgtm:

> Reviewed 6 of 6 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/27951#-:-LIM-MWcAV4jNM__K79Y:bnfp4nl)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)

___
*[pkg/sql/distsqlrun/server.go, line 341 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/27951#-LILzuYjEK2oKQ9gOLFH:-LILzuYjEK2oKQ9gOLFI:b-8qq5w6) ([raw file](https://github.com/cockroachdb/cockroach/blob/d7351ba97e3690d4975019b51bcc50dd2095ed41/pkg/sql/distsqlrun/server.go#L341)):*
> ```Go
> 	}
> 
> 	var be sessiondata.BytesEncodeFormat
> ```

You do this more than once.  DRY it up?
___
*[pkg/sql/distsqlrun/server.go, line 360 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/27951#-LIM-2Gz46ak0dfmOiJD:-LIM-2H-6wPjFESiO4iN:bhzqscy) ([raw file](https://github.com/cockroachdb/cockroach/blob/d7351ba97e3690d4975019b51bcc50dd2095ed41/pkg/sql/distsqlrun/server.go#L360)):*
> ```Go
> 		SearchPath:        sessiondata.MakeSearchPath(req.EvalContext.SearchPath),
> 		SequenceState:     sessiondata.NewSequenceState(),
> 		BytesEncodeFormat: be,
> ```

If you're alphabetizing location, why not move bytesencodeformat as well.
___
*[pkg/sql/logictest/testdata/logic_test/bytes, line 122 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/27951#-LIM-BiEEWKgWMtCFiR7:-LIM-BiEEWKgWMtCFiR8:b-8d1d7q) ([raw file](https://github.com/cockroachdb/cockroach/blob/d7351ba97e3690d4975019b51bcc50dd2095ed41/pkg/sql/logictest/testdata/logic_test/bytes#L122)):*
> ```
> SELECT e'a\\01'::STRING::BYTEA
> 
> # Regression test for #27950.
> ```

make this a subtest and drop the table at the end of it

File: pkg/sql/distsqlrun/api.proto [hunk 1]
```diff
@@ -63,6 +63,14 @@ message EvalContext {
   optional string user = 7 [(gogoproto.nullable) = false];
   optional SequenceState seq_state = 8 [(gogoproto.nullable) = false];
   optional string application_name = 9 [(gogoproto.nullable) = false];
+  optional BytesEncodeFormat bytes_encode_format = 10 [(gogoproto.nullable) = false];
+}
+
+// BytesEncodeFormat is the configurationf or bytes to string conversions.
```
Comment (jordanlewis) [inline pkg/sql/distsqlrun/api.proto]: typo

**Review by knz — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/27951)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)

___
*[pkg/sql/distsqlrun/server.go, line 341 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/27951#-LILzuYjEK2oKQ9gOLFH:-LIM101ZDOYx4pRDyABE:bii1e83) ([raw file](https://github.com/cockroachdb/cockroach/blob/d7351ba97e3690d4975019b51bcc50dd2095ed41/pkg/sql/distsqlrun/server.go#L341)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary><blockquote>

You do this more than once.  DRY it up?
</blockquote></details>

Nope, these are two different patterns (conversion in opposite directions).
___
*[pkg/sql/distsqlrun/server.go, line 360 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/27951#-LIM-2Gz46ak0dfmOiJD:-LIM13ucDp8CpoUPuqB7:bxww1i2) ([raw file](https://github.com/cockroachdb/cockroach/blob/d7351ba97e3690d4975019b51bcc50dd2095ed41/pkg/sql/distsqlrun/server.go#L360)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary><blockquote>

If you're alphabetizing location, why not move bytesencodeformat as well.
</blockquote></details>

Nope, because the follow-up PR #27952 rewrites this part altogether.
___
*[pkg/sql/logictest/testdata/logic_test/bytes, line 122 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/27951#-LIM-BiEEWKgWMtCFiR7:-LIM1iurEUnxjP4xlS5Y:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/d7351ba97e3690d4975019b51bcc50dd2095ed41/pkg/sql/logictest/testdata/logic_test/bytes#L122)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary><blockquote>

make this a subtest and drop the table at the end of it
</blockquote></details>

Done.

**Review by jordanlewis — APPROVED** [review]

File: pkg/sql/distsqlrun/api.proto (hunk 1 above)
Comment (knz) [reply pkg/sql/distsqlrun/api.proto]: Good catch! Fixed.

**Pull request merged**